lib/lower/Bridge.cpp

```diff
@@ -661,83 +661,33 @@ class FirConverter : public AbstractConverter {
     // Helper to get address and length from an Expr that is a character
     // variable designator
     auto getAddrAndLength{[&](const SomeExpr &charDesignatorExpr)
-                              -> std::pair<mlir::Value, mlir::Value> {
+                              -> CharacterOpsCreator::CharValue {
       M::Value addr = genExprAddr(charDesignatorExpr);
       const auto &charExpr{
           std::get<Ev::Expr<Ev::SomeCharacter>>(charDesignatorExpr.u)};
       std::optional<Ev::Expr<Ev::SubscriptInteger>> lenExpr{charExpr.LEN()};
       assert(lenExpr && "could not get expression to compute character length");
-      M::Value len = genExprValue(Ev::AsGenericExpr(std::move(*lenExpr)));
-      return {addr, len};
+      M::Value len{genExprValue(Ev::AsGenericExpr(std::move(*lenExpr)))};
+      return CharacterOpsCreator::CharValue{addr, len};
     }};
```

Comment:
Would this be useful anywhere else? It seems like it might be, and then it should be a utility member function in the class rather than a local lambda.

Reply:
Probably, but until I have other use case I am not sure where it would be best for it to live. The `CharacterOpsCreator` is not good for it because I do not think it should depend on `Expr<...>`.